torch/testing/_internal/common_methods_invocations.py

```diff
@@ -1200,6 +1200,39 @@ def sample_inputs_svd(op_info, device, dtype, requires_grad=False):
 def sample_inputs_linalg_svd(op_info, device, dtype, requires_grad=False):
     return _sample_inputs_svd(op_info, device, dtype, requires_grad, is_linalg_svd=True)
 
+def sample_inputs_eig(op_info, device, dtype, requires_grad=False):
+    eigvecs = make_tensor((S, S), device=device, dtype=dtype,
+                          low=None, high=None)
+    eigvals = make_tensor((S,), device=device, dtype=dtype,
+                          low=None, high=None)
+    input = (eigvecs * eigvals.unsqueeze(-2)) @ eigvecs.inverse()
+    input.requires_grad_(requires_grad)
+
+    def process_input(eigpair):
+        eigvals, eigvecs = eigpair
+        if dtype.is_complex:
+            # eig produces eigenvectors which are normalized to 1 norm.
+            # Note that if v is an eigenvector, so is v * e^{i \phi},
+            # and |v| = |v * e^{i \phi}| = 1.
+            # This, however, makes the eigenvector computation process
+            # rather unstable under a perturbation that gradcheck is using,
+            # resulting in failures of both torch.autograd.gradcheck and
+            # torch.autograd.gradgradcheck.
```

Comment:
I think this comment is misleading. Finite differences used by gradcheck computes approximately correct answer, it's the analytical derivative implemented in `eig_backward` is not correct if the loss function value is not constant under sign/phase change of eigenvectors, that's why `abs` is needed to make the loss function "gauge-invariant".